lib/terminus.js

```diff
@@ -57,33 +57,37 @@ function decorateWithHealthCheck (server, options) {
 }
 
 function decorateWithSignalHandler (server, options) {
-  const { signal, onSignal, beforeShutdown, onShutdown, timeout, logger } = options
+  const { signals, onSignal, beforeShutdown, onShutdown, timeout, logger } = options
 
   stoppable(server, timeout)
 
   const asyncServerStop = promisify(server.stop).bind(server)
 
-  function cleanup () {
-    state.isShuttingDown = true
-
-    beforeShutdown()
-      .then(() => asyncServerStop())
-      .then(() => onSignal())
-      .then(() => onShutdown())
-      .then(() => {
-        process.removeListener(signal, cleanup)
-        process.kill(process.pid, signal)
-      })
-      .catch((error) => {
-        logger('error happened during shutdown', error)
-        process.exit(1)
+  function cleanup (signal) {
+    if(!state.isShuttingDown) {
+      state.isShuttingDown = true
+      beforeShutdown()
+        .then(() => asyncServerStop())
+        .then(() => onSignal())
+        .then(() => onShutdown())
+        .then(() => {
+          signals.forEach(sig => process.removeListener(sig, cleanup))
+          process.kill(process.pid, signal)
+        })
+        .catch((error) => {
+          logger('error happened during shutdown', error)
+          process.exit(1)
       })
+    }
   }
-  process.on(signal, cleanup)
+  signals.forEach(
```

Comment:
should we merge `signal` into `signals`, so the listener is only appended once?

Reply:
`signal` gets merged into the `signals` array on L107, just before the call to `decorateWithSignalHandler`

Reply:
oh cool, thanks - I missed that